vllm/distributed/utils.py

```diff
@@ -105,8 +105,10 @@ def gpu_p2p_access_check(i: int, j: int) -> bool:
     cuda_visible_devices = os.environ.get("CUDA_VISIBLE_DEVICES", None)
     if cuda_visible_devices is None:
         cuda_visible_devices = ",".join(str(i) for i in range(num_dev))
-    path = os.path.expanduser(
-        f"~/.config/vllm/gpu_p2p_access_cache_for_{cuda_visible_devices}.json")
+    cache_dir = os.environ.get(
```

Comment:
now i see it, looks like the whole logic is not even needed, we can just populate `_gpu_p2p_access_cache`?

Reply:
What do you mean by `the whole logic is not even needed` ?

Reply:
I meant, we don't need to dump a json to disk, instead we just populate `_gpu_p2p_access_cache` if it's `None`, otherwise, we just read from `_gpu_p2p_access_cache`. Am i missing anything?

Reply:
save to disk is useful to save memory, unless you want the first rank of the worker to have less gpu memory: https://github.com/vllm-project/vllm/pull/4021 .